include/bluetooth/hci.h

```diff
@@ -1938,6 +1938,15 @@ typedef bool bt_hci_vnd_evt_cb_t(struct net_buf_simple *buf);
   */
 int bt_hci_register_vnd_evt_cb(bt_hci_vnd_evt_cb_t cb);
 
+/** @brief Get remote version info
+ *
+ *  @param conn Connection object.
+ *  @param rv_cb Callback that will be called with the remote version.
+ *
+ *  @return Zero on success or (negative) error code on failure.
+ */
+int bt_conn_remote_version(struct bt_conn *conn, bt_remote_version_cb *rv_cb);
```

Comment:
Sorry, I should have said this earlier when you hinted at moving this here: the right place is `conn.h` as the name of the function also suggests (no "hci" in it). The Zephyr Bluetooth API is designed to be independent from HCI (with the exception of `hci.h`) in order to allow replacing the stack with a potentially non-HCI based one (e.g. the stack running on a separate chip with some high-level interface to it).

Reply:
So I undo the removal of the function and just shorten it in the manner you described earlier?
Or do you have another suggestion?

Reply:
Just move the declaration from hci.h to conn.h. No other change needed.

Reply:
The function is in hci_core.c, don't we need something else in order to have it in conn.h?

Reply:
Oh, then conn.c us more appropriate, but the header location is the most important part.

Reply:
The functions need to be in the same file since they share a semaphore, so I should move both functions to conn.c? Or should i just leave it as is and just have the header reference in conn.h with the functions in hci_core.c?